src/bundle/Resources/public/js/CKEditor/embed/content-inline/embed-inline-editing.js

```diff
@@ -0,0 +1,153 @@
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+import { toWidget } from '@ckeditor/ckeditor5-widget/src/utils';
+import Widget from '@ckeditor/ckeditor5-widget/src/widget';
+
+import EmbedContentInlineCommand from './embed-inline-command';
+
+class EmbedContentInlineEditing extends Plugin {
+    static get requires() {
+        return [Widget];
+    }
+
+    defineSchema() {
+        const schema = this.editor.model.schema;
+
+        schema.register('embedInline', {
+            isObject: true,
+            isInline: true,
+            allowWhere: '$text',
+            allowAttributes: ['contentId', 'contentName'],
+        });
+    }
+
+    defineConverters() {
+        const conversion = this.editor.conversion;
+
+        conversion
+            .for('editingDowncast')
+            .elementToElement({
+                model: 'embedInline',
+                view: (modelElement, { writer: downcastWriter }) => {
+                    const container = downcastWriter.createContainerElement('span', {
+                        'data-ezelement': 'ezembedinline',
+                        'data-ezview': 'embed-inline',
+                        class: 'ez-embed-inline',
+                    });
+                    const preview = downcastWriter.createUIElement('span', { class: 'ez-embed-content' }, function(domDocument) {
+                        const domElement = this.toDomElement(domDocument);
+
+                        domElement.innerHTML = `<svg class="ez-icon ez-icon--medium ez-icon--secondary">
+                        <use xlink:href="${window.eZ.helpers.icon.getIconPath('embed')}"></use>
+                        <span class="ez-embed-content__title">${modelElement.getAttribute('contentName')}</span>
+                    </>`;
+
+                        return domElement;
+                    });
+
+                    downcastWriter.insert(downcastWriter.createPositionAt(container, 0), preview);
+
+                    return toWidget(container, downcastWriter);
+                },
+            })
+            .add((dispatcher) =>
+                dispatcher.on('attribute:contentName', (event, data, conversionApi) => {
+                    const downcastWriter = conversionApi.writer;
+                    const modelElement = data.item;
+                    const viewElement = conversionApi.mapper.toViewElement(modelElement);
+                    const preview = downcastWriter.createUIElement('span', { class: 'ez-embed-content' }, function(domDocument) {
```

Comment:
maybe fat arrowy?

Reply:
cannot be an arrow function, the `this` matters.